### PR TITLE
Moen 20813 push Optin Inpps

### PR DESCRIPTION
**Pull request by kkgowtham-engg-sdk**
### Jira Ticket
MOEN-20813
https://moengagetrial.atlassian.net/browse/MOEN-20813

### Description
Support for 2Step Push OptIn In Apps

### CODE REVIEW

File: core/lib/constants.dart [hunk 1]
```diff
@@ -30,6 +30,8 @@ const String methodSetupNotificationChannelAndroid =
     "setupNotificationChannels";
 const String methodNavigateToSettingsAndroid = "navigateToSettings";
 const String methodRequestPushPermissionAndroid = "requestPushPermission";
+const String updatePushPermissionRequestCount =
```
Comment (arshiya-moengage) [inline core/lib/constants.dart:33]: the field name should have `method` prefix

File: core/lib/constants.dart [hunk 2]
```diff
@@ -132,3 +134,5 @@ const String keyAdId = "isAdIdTrackingEnabled";
 // permission
 const String keyIsPermissionGranted = "isGranted";
 const String keyPermissionType = "type";
+
+const String updatePushPermissionCountKey = "pushOptinInAttemptCount";
```
Comment (arshiya-moengage) [inline core/lib/constants.dart:138]: follow the naming conventions, the name should be `keyUpdatePushPermissionCount` instead `updatePushPermissionCountKey`

File: core/lib/moe_android_core.dart [hunk 3]
```diff
@@ -214,6 +214,12 @@ class MoEAndroidCore {
     return json.encode(payload);
   }
 
+  String _getUpdatePushCountJsonPayload(int requestCount, String appId) {
```
Comment (arshiya-moengage) [inline core/lib/moe_android_core.dart:217]: Don't add new APIs in the middle of the file, move this to bottom

File: core/android/src/main/kotlin/com/moengage/flutter/EventEmitterImpl.kt [hunk 4]
```diff
@@ -150,6 +152,7 @@ class EventEmitterImpl(private val onEvent: (methodName: String, payload: String
             eventMap[EventType.INAPP_SELF_HANDLED_AVAILABLE] = "onInAppSelfHandle"
             eventMap[EventType.PUSH_TOKEN_GENERATED] = "onPushTokenGenerated"
             eventMap[EventType.PERMISSION] = "onPermissionResult"
-         }
+            eventMap[EventType.REQUEST_NOTIFICATION_PERMISSION] = "requestNotificationPermission"
```
[the remark this answers is not in the document]
Comment (kkgowtham-engg-sdk) [reply core/android/src/main/kotlin/com/moengage/flutter/EventEmitterImpl.kt]: Removed

File: core/CHANGELOG.md [hunk 5]
```diff
@@ -1,8 +1,13 @@
 # MoEngage Flutter Plugin
 
+## Next Release
+
+- Support for 2Step Push Optin InApps
```
[the remark this answers is not in the document]
Comment (kkgowtham-engg-sdk) [reply core/CHANGELOG.md]: Fixed

File: core/lib/constants.dart (hunk 1 above)
Comment (kkgowtham-engg-sdk) [reply core/lib/constants.dart]: Updated

File: core/lib/constants.dart (hunk 2 above)
Comment (kkgowtham-engg-sdk) [reply core/lib/constants.dart]: Updated

File: core/lib/moe_android_core.dart (hunk 3 above)
Comment (kkgowtham-engg-sdk) [reply core/lib/moe_android_core.dart]: Resolved

**Review by arshiya-moengage — APPROVED** [review]

**Pull request merged**